### PR TITLE
Resume each batch from the last when trimming flow sessions

### PR DESCRIPTION
**Pull request by rowanseymour**
## Summary

`trim_flow_sessions` deleted in batches by re-running the same `SELECT ... WHERE ended_on <= ? LIMIT n` for every batch. Rows deleted by earlier batches stay in the `flowsessions_ended` index until vacuum removes them, so each batch walked past every entry deleted so far and a large trim became quadratic. The task could then outlive its cron lock and fail with `LockNotOwnedError` when it tried to release it.

## Changes

- **`temba/flows/tasks.py`**: `trim_flow_sessions` now orders batches by `ended_on` and starts each one at the last `ended_on` the previous batch deleted, so it no longer re-walks entries it has already deleted. The filter is inclusive so sessions sharing the boundary value aren't skipped, and every batch deletes at least one row so it always makes progress.
- **`temba/flows/tests/test_session.py`**: runs the trim with a small batch size, with sessions sharing an `ended_on` that fall across batch boundaries.

The logic lives in the task rather than the generic `delete_in_batches`. None of that helper's other callers has an index that would benefit, and ordering them would add a sort to every batch.

## Test plan

- [x] Flow session trim test passes, including batches split across shared `ended_on` values
- [x] `code_check.py` passes
- [x] Local `EXPLAIN` of the batch query uses `flowsessions_ended`

### CODE REVIEW

File: temba/utils/models/base.py [hunk 1]
```diff
@@ -25,17 +25,29 @@ def patch_queryset_count(qs, function):
...
+    entries of rows already deleted but not yet vacuumed - which makes large deletes quadratic.
     """
 
     num_deleted = 0
+    resume_from = None
 
     while True:
-        pk_batch = list(qs.values_list(pk, flat=True)[:batch_size])
+        if order_by:
+            batch_qs = qs.filter(**{f"{order_by}__gte": resume_from}) if resume_from is not None else qs
```
[the remark this answers is not in the document]
Comment (rowanseymour) [reply temba/utils/models/base.py]: No longer applicable: the resumable batching now lives only in the flow session trim task, where `ended_on` is never NULL, and `delete_in_batches` is unchanged.
